SmartyadsBidAdapter remove usprivacy

rishko00 opened the pull request:
<!--
Thank you for your pull request! 

Please title your pull request like this: 'Module: Change', eg 'Fraggles Bid Adapter: support fragglerock'

Please make sure this PR is scoped to one change or you may be asked to resubmit. 
 
Please make sure any added or changed code includes tests with greater than 80% code coverage. 

See https://github.com/prebid/Prebid.js/blob/master/CONTRIBUTING.md#testing-prebidjs for documentation on testing Prebid.js.

For any user facing change, submit a link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/
-->

## Type of change
<!-- Remove items that don't apply and/or select an item by changing [ ] to [x] -->
- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter  <!--  IMPORTANT: also submit your bidder parameter documentation as noted in https://docs.prebid.org/dev-docs/bidder-adaptor.html#submitting-your-adapter -->
- [ ] Updated bidder adapter  <!--  IMPORTANT: (1) consider whether you need to upgrade your bidder parameter documentation in https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders and (2) if you have a Prebid Server adapter, please consider whether that should be updated as well. --> 
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
<!-- Describe the change proposed in this pull request -->

<!-- For new bidder adapters, please provide the following
- contact email of the adapter’s maintainer
- test parameters for validating bids:
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page. -->


## Other information
<!-- References to related PR or issue #s, @mentions of the person or team responsible for reviewing changes, etc. -->

patmmccann commented on modules/smartyadsBidAdapter.js:
```diff
@@ -79,19 +79,19 @@ export const spec = {
-      'secure': 1,
-      'host': location?.domain ?? '',
-      'page': location?.page ?? '',
       'coppa': config.getConfig('coppa') === true ? 1 : 0,
```
We will not be accepting tricks to fill the duplication check

rishko00 replied:
@patmmccann OK, but how can I fix that? I saw the report and duplication check marks, for example, the isBidResponseValid function is used in lots of adapters and looks almost the same.

patmmccann replied:
Check out this work in progress from an adapter structured very similar to yours

patmmccann replied:
https://github.com/MaksymTeqBlaze/Prebid.js/commit/0249eb7a258d610e937a114676e24c1d3f6c956e

rishko00 replied:
@patmmccann I'm sorry, but I still don't get it. The previous version of the isBidResponseValid function was flagged as duplicate code by your tool, but it doesn't need any updates, right? And the same function with same code is used in other adapters

patmmccann replied:
You and other sites have duplicate code, we need you to move common functions into libraries before making further updates to your adapter

patmmccann replied:
#8527 has an example on how to move common functions into a library

patmmccann reviewed (changes requested):
Please do not attempt malicious changes designed to circumvent our rules

- [

rishko00 commented:
@patmmccann, I've moved some functions to the libs and removed some unused fields from the request and unused functions. However, there are still common functions, such as isBidResponseValid and interpretResponse. If I move isBidResponseValid to libs, there will still be an issue with duplicated code, but I can't rewrite it because it's used for common validations. Also I didn't find example of adapter where those functions are working properly. I need your advice on how to proceed.

patmmccann commented:
Hi @rishko00 take a look at https://github.com/MaksymTeqBlaze/Prebid.js/commit/0249eb7a258d610e937a114676e24c1d3f6c956e#diff-a442c024395533115322fcbbfe796836b8ea37974565360dda15e9c93fbca181 or at #11849 to see examples of bidders merging common functions. The library function you added looks great but I am not sure if it would be imported in multiple adapters.

patmmccann commented:
#11854 has merged in and is a great example

rishko00 commented:
@patmmccann yes, got it, I've pushed changes

patmmccann commented:
Looks great! Can you undo your changes to render files and we're good, thanks!!
